library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetPrompt.java

```diff
@@ -887,6 +889,13 @@ else if (event.getAction() == KeyEvent.ACTION_UP
             return super.dispatchKeyEventPreIme(event);
         }
 
+        @Override
+        protected void onDetachedFromWindow()
+        {
+            super.onDetachedFromWindow();
+            mPrompt.cleanUpAnimation();
```

Comment:
Is there a reason that this isn't a call to `cleanUpPrompt`?

Reply:
`cleanUpPrompt` also notifies state changes which seems like we'd be introducing a race condition... I just realized I didn't share the leak with you, it's just the animator stuff which is leaking:
```
In com.supercilex.robotscouter.debug:2.3.1-DEBUG:1.
* com.supercilex.robotscouter.HomeActivity has leaked:
* GC ROOT android.widget.Toast$TN.mNextView
* references android.widget.LinearLayout.mParent
* references android.view.ViewRootImpl.mChoreographer
* references android.view.Choreographer.mCallbackQueues
* references array android.view.Choreographer$CallbackQueue[].[1]
* references android.view.Choreographer$CallbackQueue.mHead
* references android.view.Choreographer$CallbackRecord.action
* references android.animation.AnimationHandler$1.this$0 (anonymous implementation of android.view.Choreographer$FrameCallback)
* references android.animation.AnimationHandler.mAnimationCallbacks
* references java.util.ArrayList.elementData
* references array java.lang.Object[].[0]
* references android.animation.ValueAnimator.mUpdateListeners
* references java.util.ArrayList.elementData
* references array java.lang.Object[].[0]
* references uk.co.samuelwall.materialtaptargetprompt.MaterialTapTargetPrompt$10.this$0 (anonymous implementation of android.animation.ValueAnimator$AnimatorUpdateListener)
* references uk.co.samuelwall.materialtaptargetprompt.MaterialTapTargetPrompt.mView
* references uk.co.samuelwall.materialtaptargetprompt.MaterialTapTargetPrompt$PromptView.mContext
* leaks com.supercilex.robotscouter.HomeActivity instance

* Retaining: 6.5 kB.
* Reference Key: 8200773e-a087-4f54-8dd7-9fe208a43681
* Device: Google google Pixel 2 XL taimen
* Android Version: 8.1.0 API: 27 LeakCanary: 1.5.4 74837f0
* Durations: watch=5315ms, gc=172ms, heap dump=1438ms, analysis=154338ms
```